odoo/addons/base/models/res_users.py

```diff
@@ -1492,6 +1543,21 @@ def unlink(self):
 class UsersView(models.Model):
     _inherit = 'res.users'
 
+    user_group_warning = fields.Text(string="User Group Warning", compute="_compute_user_group_warning")
+
+    @api.depends('groups_id')
+    @api.depends_context('show_user_group_warning')
+    def _compute_user_group_warning(self):
+        self.user_group_warning = False
+        if self._context.get('show_user_group_warning'):
+            user_group_id = self.env['ir.model.data']._xmlid_to_res_id('base.group_user')
+            # Sometimes Share user domain is not working due to cache. so, have to get an internal users manually.
+            for user in self.filtered_domain([('groups_id', 'in', [user_group_id])]):
+                group_inheritance_warnings = user.groups_id._prepare_warning_for_group_inheritance(user)
+
+                if group_inheritance_warnings:
+                    user.user_group_warning = "\n".join(group_inheritance_warnings)
```

Comment:
why not doing the `"\n".join` in `_prepare_warning_for_group_inheritance` directly ?